### PR TITLE
refactor/Object `transformer` package

### PR DESCRIPTION
**Pull request by carpawell**

### CODE REVIEW

File: pkg/core/object/target.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package transformer
+package object
```
Comment (cthulhu-rider) [inline pkg/core/object/target.go:1]: correct, but how about locate package in the `pkg/services/object/internal` if it's not intended to be used outside Object service?

File: pkg/services/object/put/slice.go [hunk 2]
```diff
@@ -21,23 +21,23 @@ type slicingTarget struct {
...
 	homoHashDisabled bool
 
-	nextTarget transformer.ObjectTarget
+	nextTarget internal.Target
 
 	payloadWriter *slicer.PayloadWriter
 }
 
-// returns transformer.ObjectTarget for raw root object streamed by the client
+// returns [objectcore.Target] for raw root object streamed by the client
```
Comment (roman-khimov) [inline pkg/services/object/put/slice.go:29]: Is it `internal` or `objectcore`?

File: pkg/services/object/put/slice.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/nspcc-dev/neofs-node/pkg/services/object_manager/transformer"
+	"github.com/nspcc-dev/neofs-node/pkg/services/object/internal"
```
Comment (roman-khimov) [inline pkg/services/object/put/slice.go:7]: Usually, `internal/smth` is used, because `internal` itself doesn't mean a lot.

Comment (roman-khimov) [reply pkg/services/object/put/slice.go:7]: It's strange enough we have a whole package just for this interface. To me it looks a lot like a part of `pkg/services/object/put`. But there it'd be internal. But then who needs an internal interface?

Comment (carpawell) [reply pkg/services/object/put/slice.go:7]: https://github.com/nspcc-dev/neofs-node/pull/2671#discussion_r1420316138 describes where i saw this interface and why

part of it is used by the get service too. i wanted to see more usages but somehow we have so many `WriteChunk` and `Write(object.Object) error` (not the `oi.Writer`) so it does not look so useful sadly. but we can continue this refactor

Comment (roman-khimov) [reply pkg/services/object/put/slice.go:7]: `get` doesn't need it. It has the thing it needs already.

Comment (carpawell) [reply pkg/services/object/put/slice.go:7]: dont you like how `HeaderWriter` is included in the `Target` and how every object interface can be placed in a single common file? every header should be written with the same interface does not matter what service does it

File: pkg/services/object/put/slice.go (hunk 2 above)
Comment (carpawell) [reply pkg/services/object/put/slice.go]: has not pushed the last commit rebase, fixed

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**